### PR TITLE
docs: add metric page

### PR DESCRIPTION
**Pull request by MariusVanDerWijden**
closes https://github.com/ethereum/go-ethereum/issues/21524

### CODE REVIEW

File: docs/_interface/metrics.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
+---
+title: Metrics
+sort_key: C
+---
+
+## Meters and Timers
+
+Note, metrics collection is disabled by default in order not to incur reporting overhead for the average user. To enable it please specify the `--metrics` flag to geth.
```
Comment (holiman) [inline docs/_interface/metrics.md:8]:
```suggestion
Note, metrics collection is disabled by default in order not to incur reporting overhead for the average user. The flag `--metrics` must therefore be used to enable the basic metrics, and the flag `--metrics.expensive` can be used to enable certain metrics that are deemed 'expensive', from a resource-consumption perspective. Examples of expensive metrics is per-packet network traffic data.  
```

File: docs/_interface/metrics.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+Geth has quite a nice logging system, capable of creating leveled log entries tagged with various parts of the system. This helps enormously during debugging to see exactly what the system is doing, what branches it's taking, etc. However, logs are not particularly useful when the system does work correctly, just not very optimally: one - or even a  handful - of logged events is not really statistically relevant, and tracing more in log files can quickly become unwieldy.
```
Comment (holiman) [inline docs/_interface/metrics.md:10]: I dunno, maybe we can skip this? It's a page about metrics, I don't think we need to justify why metrics are different/better than logs.

File: docs/_interface/metrics.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+To that extent, Geth currently implement two types of metrics:
+ * **Meters**: Analogous to physical meters (electricity, water, etc), they are capable of measuring the *amount* of "things" that pass through and at the *rate* at which they do that. A meter doesn't have a specific unit of measure (byte, block, malloc, etc), it just counts arbitrary *events*. At any point in time it can report:
+   * *Total number of events* that passed through the meter
+   * *Mean throughput rate* of the meter since startup (events / second)
+   * *Weighted throughput rate* in the last *1*, *5* and *15* minutes (events / second)
+     * (*"weighted" means that recent seconds count more that in older ones*)
+ * **Timers**: Extension of *meters*, where not only the occurrence of some event is measured, its *duration* is also collected. Similarly to meters, a timer can also measure arbitrary events, but each requires a duration to be assigned individually. Beside **all** the reports a meter can generate, a timer has additionally:
```
Comment (holiman) [inline docs/_interface/metrics.md:20]: I always get this wrong, but there's this thing called `gauge` too. Is that a `timer`? If so, could you add e..g. "Timers (also called gauges".

File: docs/_interface/metrics.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+     * Percentile 80: general performance (these should be optimised)
+     * Percentile 95: worst case outliers (rare, just handle gracefully)
+
+## Creating and updating metrics
+
+Metrics can be added easily in the code:
+
+```go
+meter := metrics.NewMeter("system/memory/allocs")
+timer := metrics.NewTimer("chain/inserts")
```
Comment (holiman) [inline docs/_interface/metrics.md:33]: If you want, you could mention the nifty construct with `NewOrRegisteredMeter("foo/baz")`, which means that different packages can use the same meters without creating dependency cycles. 

But maybe that is overkill

**Review by DGKSK8LIFE — COMMENTED** [review]
nice

**Pull request merged**